instrumentation-api-caching/build.gradle

```diff
@@ -41,3 +41,11 @@ jar {
 
   dependsOn shadowJar
 }
+
+publishing {
```

Comment:
Here `configurations` doesn't include runtimeclasspath which is why it's not in `java["components]` (I guess...)

Reply:
This is needed here because we disable `jar` task.

I don't exactly like how this module is now published/used. I mean when `instrumentation-api` module declares dependency on `api project(":instrumentation-api-caching")` what exactly does it want? Just jar (which is disabled) or full shadow jar? If the latter we should do this more explicit.

Reply:
Yeah I think we could consider the technique we found for jctools in SDK

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace-shaded-deps/build.gradle.kts
https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/build.gradle.kts#L73

We don't publish the shadowed thing separately but embed it into the SDK's normal JAR, so here we could embed into instrumentation-api.